source/containers.tex

```diff
@@ -2888,18 +2889,18 @@
   class deque {
   public:
     // types:
-    typedef value_type&                                         reference;
-    typedef const value_type&                                   const_reference;
-    typedef @\impdefnc@                              iterator;        // see \ref{container.requirements}
-    typedef @\impdefnc@                              const_iterator;  // see \ref{container.requirements}
-    typedef @\impdefnc@                              size_type;       // see \ref{container.requirements}
-    typedef @\impdefnc@                              difference_type; // see \ref{container.requirements}
-    typedef T                                                   value_type;
-    typedef Allocator                                           allocator_type;
-    typedef typename allocator_traits<Allocator>::pointer       pointer;
-    typedef typename allocator_traits<Allocator>::const_pointer const_pointer;
-    typedef std::reverse_iterator<iterator>                     reverse_iterator;
-    typedef std::reverse_iterator<const_iterator>               const_reverse_iterator;
+    using value_type             = T;
+    using allocator_type         = Allocator;
+    using pointer                = typename allocator_traits<Allocator>::pointer;
+    using const_pointer          = typename allocator_traits<Allocator>::const_pointer;
+    using reference              = value_type&;
+    using const_reference        = const value_type&;
+    using size_type              = @\impdefnc@; // see \ref{container.requirements}
```

Comment:
If I understand correctly, you are saying replace '\impdefnc' with '\impdef'?
If so, is that a general rule, or applies only to header synopses and other code snippets?